crates/networking/rpc/rogue/transaction.rs

```diff
@@ -0,0 +1,231 @@
+            .unwrap_or_default();
+        let gas_price_request = GasPrice {}.handle(context.clone())?;
+        let max_fee_per_gas = u64::from_str_radix(
+            gas_price_request
+                .as_str()
+                .unwrap_or("0x0")
+                .strip_prefix("0x")
+                .unwrap(),
+            16,
+        )
+        .map_err(|err| RpcErr::Internal(err.to_string()))?;
+
+        let mut tx = if let Some(auth_list) = &self.authorization_list {
+            SendRawTransactionRequest::EIP7702(EIP7702Transaction {
+                chain_id,
+                to: self.to,
+                value: U256::zero(),
+                data: self.data.clone(),
+                access_list: Vec::new(),
+                authorization_list: auth_list.clone(),
+                ..Default::default()
+            })
+        } else {
+            SendRawTransactionRequest::EIP1559(EIP1559Transaction {
+                chain_id,
+                to: TxKind::Call(self.to),
+                value: U256::zero(),
+                data: self.data.clone(),
+                access_list: Vec::new(),
+                ..Default::default()
+            })
+        };
+
+        let mut generic = match tx.to_transaction() {
+            ethrex_common::types::Transaction::EIP1559Transaction(tx) => {
+                GenericTransaction::from(tx)
+            }
+            ethrex_common::types::Transaction::EIP7702Transaction(tx) => {
+                GenericTransaction::from(tx)
+            }
+            _ => unreachable!("This should never happen, we are creating the tx"),
+        };
+        generic.gas = None;
+
+        let estimate_gas_request = EstimateGasRequest {
+            transaction: generic,
+            block: None,
+        }
+        .handle(context.clone())?;
+        let gas_limit = u64::from_str_radix(
+            estimate_gas_request
+                .as_str()
+                .unwrap_or("0x0")
+                .strip_prefix("0x")
+                .unwrap(),
+            16,
+        )
+        .unwrap();
+        if gas_limit == 0 || gas_limit > GAS_LIMIT_HARD_LIMIT {
+            return Err(RpcErr::InvalidRogueMessage("tx too expensive".to_string()));
+        }
+        match tx {
+            SendRawTransactionRequest::EIP7702(ref mut tx) => {
+                tx.gas_limit = gas_limit;
+                tx.max_fee_per_gas = max_fee_per_gas;
+                tx.max_priority_fee_per_gas = max_priority_fee_per_gas;
+                tx.nonce = nonce;
+                tx.sign_inplace(&sponsor_pk);
+            }
+            SendRawTransactionRequest::EIP1559(ref mut tx) => {
+                tx.gas_limit = gas_limit;
+                tx.max_fee_per_gas = max_fee_per_gas;
+                tx.max_priority_fee_per_gas = max_priority_fee_per_gas;
+                tx.nonce = nonce;
+                tx.sign_inplace(&sponsor_pk);
+            }
+            _ => unreachable!("This should never happen, we are creating the tx"),
```

Comment:
Let's return an error here instead, you never know what will happen if we ever introduce a bug reach this path.